pymbolic/mapper/__init__.py

```diff
@@ -1063,11 +1063,12 @@ def map_common_subexpression(self, expr, *args):
         except AttributeError:
             ccd = self._cse_cache_dict = {}
 
+        key = (expr, *args)
         try:
-            return ccd[(expr, *args)]
+            return ccd[key]
         except KeyError:
             result = self.map_common_subexpression_uncached(expr, *args)
-            ccd[(expr, *args)] = result
+            ccd[key] = result
             return result
```

Comment:
This looks like a bug in ruff: doing `ccd[expr, *args]` isn't allowed in Python 3.8, but they still flag it as an issue..